### DISCUSSION TITLE
question about KafkaConnector CRD behaviour

### DISCUSSION THREAD

**Discussion by elvis-cai** [strimzi/strimzi-kafka-operator#7141] [General]
Hi, when using KafkaConnect to create kafka connect clusters, we could use KafkaConnector to manage different plugins and configs as a single source of truth, at the same time, is it possible we could use rest API to manage those plugins not declared in KafkaConnector CRD? 
Or we need create two kafka conncet clusters, one managed by KafkaConnector, another just use REST API only?
Thanks.

**Comment by scholzj** [comment]
No, you need to choose one or the other. If you use the Connector operator, the KafkaConnector CRs will be the single source of truth. That does not mean only that it is a source of truth for the configuration of a particular connector. But it means that if the KafkaConnector CR does not exist in Kubernetes, it should nto exist in Kafka Connect either.

You can of course have two separate Connect clusters one using the REST API and the other one using the Connector operator. Just don't forget to follow [this](https://strimzi.io/docs/operators/latest/full/deploying.html#con-kafka-connect-multiple-instances-str) to make sure they are really independent.

**Reply by elvis-cai** [reply]
fair enough, thanks for the quick reply. 👍
